enable/examples/demo/enable/sierpinski_zoom.py

```diff
@@ -0,0 +1,203 @@
+            # points to make recursive calls at
+            #
+            #       /\
+            #     ?/__\
+            #    X/_\/_\
+            #    /\    /\
+            #  ?/__\ ?/__\
+            #  /_\/_\/_\/_\
+            #
+
+            rel_to_point_locs = (size/2)*np.array([
+                [-1, -SQRT3],
+                [1, SQRT3],
+                [3, -SQRT3]
+            ])
+            # absolute location of those next points
+            abs_points = point + rel_to_point_locs
+
+            for point in abs_points:
+                self.add_triangle_to_path(
+                    path,
+                    point,
+                    size
+                )
+                self.sierpinski(
+                    path, point, n+1
+                )
+
+    def add_triangle_to_path(self, path, point, size):
+        """ Adds an inverted triangle to the input path, of side length size,
+        using the input point as the upper left corner of the triangle.
+        """
+        x, y = point
+        path.move_to(x, y)
+        path.line_to(x + size, y)
+        path.line_to(x + .5 * size, y - (SQRT3 / 2) * size)
+        path.line_to(x, y)
+
+
+class Viewer(HasTraits):
+
+    iterations = Range(0, 'max_iters')
+
+    triangle = Instance(AbstractOverlay)
+
+    comp = Instance(Component)
+
+    base_width = Int(500)
+
+    max_iters = Property(observe="base_width, viewport.zoom")
+
+    viewport = Instance(Viewport)
+
+    def _get_bounds(self):
+        return [self.base_width, self.base_width*(SQRT3/2)]
+
+    def _get_max_iters(self):
+        factor = 1
+        if self.viewport:
+            factor = self.viewport.zoom
+        return int(np.log(4/(factor*self.base_width))/np.log(.5) + 1)
+
+    def _comp_default(self):
+        canvas = Canvas(draw_axes=True, bgcolor="gray")
+        self.triangle = SierpinskiTriangle(
+            canvas,
+            position=[0.0, 0.0],
+            bounds=[self.base_width, self.base_width*(SQRT3/2)],
+            iterations=self.iterations,
+            max_iters=self.max_iters,
+            base_width=self.base_width,
+        )
+        canvas.overlays.append(self.triangle)
+
+        self.viewport = Viewport(
+            component=canvas, enable_zoom=True, stay_inside=True
+        )
+        self.viewport.zoom_tool.min_zoom = 1.0
+        self.viewport.tools.append(
+            ViewportPanTool(self.viewport, drag_button="right")
+        )
+
+        scrolled = Scrolled(
+            canvas,
+            inside_padding_width=0,
+            mousewheel_scroll=False,
+            viewport_component=self.viewport,
+            always_show_sb=True,
+            continuous_drag_update=True,
+        )
+        return scrolled
+
+    @observe("base_width")
+    def _update_base_width(self, event):
+        self.triangle.bounds = [self.base_width, self.base_width*(SQRT3/2)]
+        self.triangle.base_width = event.new
```

Comment:
It reads a little weird to mix `self.base_width` and `event.new` in these two lines